tests/common.py

```diff
@@ -1218,6 +1218,8 @@ async def mock_async_load(
         if store._data is None:
             # No data to load
             if store.key not in data:
+                # Make sure the next attempt will still load
+                store._load_task = None
```

Comment:
If we get into this block the real `_async_load` is never run which means the task will never be cleared

https://github.com/home-assistant/core/blob/62e518badb529593bd0a98f573d001393493ab3d/homeassistant/helpers/storage.py#L133